### PR TITLE
fix: replace mockresourcemanager generated Project number

### PR DESCRIPTION
**Pull request by yuwenma**
This is a follow up PR to https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1934 and blocks https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2061

We need to fix the existing golden log to turn on the stricter PRESUBMIT check.

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/project/projectmovedfoldertofolder/_generated_object_projectmovedfoldertofolder.golden.yaml [hunk 1]
```diff
@@ -24,5 +24,5 @@ status:
     reason: UpToDate
     status: "True"
     type: Ready
-  number: "2589592254"
+  number: "518915279"
```
Comment (justinsb) [inline pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/project/projectmovedfoldertofolder/_generated_object_projectmovedfoldertofolder.golden.yaml:27]: This isn't a magic number, is it?

File: tests/e2e/normalize.go [hunk 2]
```diff
@@ -95,9 +95,18 @@ func normalizeKRMObject(u *unstructured.Unstructured, project testgcp.GCPProject
 	visitor.stringTransforms = append(visitor.stringTransforms, func(path string, s string) string {
 		return strings.ReplaceAll(s, project.ProjectID, "${projectId}")
 	})
+
+	// Update the project number with variable marker.
+	if u.GetKind() == "Project" {
+		// The ProjectNumber is ProjectID based. For those tests relying on the Mock Resource Manager server,
+		// the ProjectID is dynamic, different from the default mock test Project "mock-project",
+		// so the number is different as well.
+		visitor.replacePaths[".status.number"] = fmt.Sprintf("%d", project.ProjectNumber)
```
Comment (justinsb) [inline tests/e2e/normalize.go:104]: Maybe we should replace with `${projectNumber}` ?  Though TBH I'm not entirely sure why it doesn't get replaced below.  Maybe the order in which these replacements run?

File: pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/project/projectmovedfoldertofolder/_generated_object_projectmovedfoldertofolder.golden.yaml (hunk 1 above)
Comment (yuwenma) [reply pkg/test/resourcefixture/testdata/basic/resourcemanager/v1beta1/project/projectmovedfoldertofolder/_generated_object_projectmovedfoldertofolder.golden.yaml:27]: not ready for review. I have a better idea.

File: tests/e2e/normalize.go (hunk 2 above)
Comment (yuwenma) [reply tests/e2e/normalize.go]: Yes, yes, I also want to use `${projectNumber}`. Please see the last commit. 
The reason of not replacing is that those tests have two Projects, one is the "mock-project", the other is the real project which `uniqId` is dynamically generated. The below replacement `strings.ReplaceAll(s, fmt.Sprintf("%d", project.ProjectNumber), "${projectNumber}")` can only find the `mock-project` number, but not the dynamic ones. This only happens for tests that call the mock resource manager.

**Review by justinsb — APPROVED** [review]

**Pull request merged**